src/python/WMCore/WMRuntime/Watchdog.py

```diff
@@ -78,6 +81,37 @@ def setupMonitors(self, task, wmbsJob):
                 #This should be a config section
                 monitorArgs = getattr(task.data.watchdog, monitor)
                 args = monitorArgs.dictionary_()
+                # Scale resources according to the HTCondor runtime environment.
+                origCores = 1
+                try:
+                    origCores = int(task.data.application.multicore.numberOfCores)
+                except AttributeError:
+                    print("No value for numberOfCores.  Assuming 1")
+                resources = {'cores': origCores}
+                origMaxRSS = args.get('maxRSS')
+                if origMaxRSS:
+                    resources['memory'] = origMaxRSS
+                # Actually parses the HTCondor runtime
+                resizeResources(resources)
+                # HTCondor doesn't explicitly scale VSize; hence we scale it
+                # according to the scaling of maxRSS between the workflow and runtime.
+                if origMaxRSS and ('maxVSize' in args):
+                    args['maxVSize'] *= resources['memory'] / origMaxRSS
```

Comment:
The problem is, I'm not sure there is a "very bad" when it comes to VSize. The application can reserve an enormous amount of memory and in 64 bit space that's probably in the EB range. But as long as it doesn't fill it with anything, RSS stays in bounds.